test/test_framework.py

```diff
@@ -612,6 +612,110 @@ class CustomObject:
 
             framework_copy.commit()
 
+    def test_comparison_operations(self):
+        class SomeObject(Object):
+            state = StoredState()
+
+            def __init__(self, framework, key):
+                super().__init__(framework, key)
+
+        test_operations = [(
+            {"1"},  # Left operand.
+            {"1", "2"},  # Right operand.
+            lambda a, b: a < b,  # An operation to test.
+            True,  # The expected result of operation(left_operand, right_operand).
+            False,  # The expected result of operation(right_operand, left_operand).
+        ), (
+            {"1"},
+            {"1", "2"},
+            lambda a, b: a > b,
+            False,
+            True
+        ), (
+            # Empty set comparison.
+            set(),
+            set(),
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            {"a", "c"},
+            {"c", "a"},
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            dict(),
+            dict(),
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            {"1": "2"},
+            {"1": "2"},
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            {"1": "2"},
+            {"1": "3"},
+            lambda a, b: a == b,
+            False,
+            False
+        ), (
+            [],
+            [],
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            [1, 2],
+            [1, 2],
+            lambda a, b: a == b,
+            True,
+            True
+        ), (
+            [1, 2, 5, 6],
+            [1, 2, 5, 8, 10],
+            lambda a, b: a <= b,
+            True,
+            False
+        ), (
+            [1, 2, 5, 6],
+            [1, 2, 5, 8, 10],
+            lambda a, b: a < b,
+            True,
+            False
+        ), (
+            [1, 2, 5, 8],
+            [1, 2, 5, 6, 10],
+            lambda a, b: a > b,
+            True,
+            False
+        ), (
+            [1, 2, 5, 8],
+            [1, 2, 5, 6, 10],
+            lambda a, b: a >= b,
+            True,
+            False
+        ), (
+            [1, 2, 5, 8],
+            [1, 2, 5, 6, 10],
+            lambda a, b: a <= b,
+            False,
+            True
+        )]
+
+        framework = self.create_framework()
+
+        for stored_operand, builtin_operand, operation, ab_res, ba_res in test_operations:
+            obj = SomeObject(framework, "1")
+            obj.state.stored_operand = stored_operand
+
+            for a, b, expected in [(obj.state.stored_operand, builtin_operand, ab_res), (builtin_operand, obj.state.stored_operand, ba_res)]:
+                result = operation(a, b)
+                self.assertEqual(result, expected)
```

Comment:
More to read doesn't necessarily mean easier to read, and using a loop doesn't necessarily improve readability, space, or performance:
```
for a, b, op, op_ab, op_ba in test_operations:
    obj.state.a = a
    self.assertEqual(op(obj.state.a, b), op_ab)
    self.assertEqual(op(b, obj.state.a), op_ba)
```

```